### ISSUE TITLE
pushUpdates & pullUpdates Do Not Work in a Docker Container

### ISSUE THREAD

**Issue by alonalbert** [mipt-npm/plotly.kt#84]
More specifically, they do not work unless the container is running in host network mode.

The problem seems to be that the PlotlyServer baseUrl is using a hostname that is internal to the docker container but the value is being passed to the JavaScript which is running in an external network.

**Comment by altavir** [comment]
Indeed, the Url is resolved from the request. I will add a custom property to PlotlyServer, which will allow to set it up manually.

**Comment by altavir** [comment]
@alonalbert I've pushed a commit that possibly addresses the issue. It allows setting override host and port (not for Jupyter). If you can check it, please let me know.

**Comment by alonalbert** [comment]
I'll check when I get a chance, though I'm not sure it will be enough.

Keep in mind that the hostname might not be a static value. For a running
server, i can access it via different hostnames. Local IP, local hostname,
hostname alias, public IP, public IP with different port etc

I suspect you need something that gets the actual hostname and IP that the
JavaScript actually is using, not a static setting.

Perhaps you already did that, i didn't look at the commit.

**Comment by altavir** [comment]
It is not static.

Look at the code here: https://github.com/mipt-npm/plotly.kt/blob/5980ffd7abddcb0a273a644d77ad69d517b01201/plotlykt-server/src/main/kotlin/space/kscience/plotly/server/PlotlyServer.kt#L187-L193

It takes the URI from the request and then transforms it. So the actual address embedded in the page is depending on where the page has been requested from. I am just not sure, how it will work with multiple interfaces. Your suggestions are welcome.
